xapi/xapi.go

```diff
@@ -152,6 +191,7 @@ func (r *partialResult) Next() (handle int64, data []types.Datum, err error) {
 
 // Close closes the sub result.
 func (r *partialResult) Close() error {
+	close(r.done)
```

Comment:
this will cause a panic. for example, if there are two goroutines, one raises a timeout error, and then close the channel when query Closes all executors. After that another goroutine also returns a err and writes it to a closed channel.